ngraph/src/ngraph/op/ctc_loss.cpp

```diff
@@ -0,0 +1,234 @@
+
+    // check shapes of input tensors
+    size_t batch_size = 1;
+    bool is_batch_size_set = false;
+    size_t time_steps = 1;
+    bool is_time_steps_set = false;
+
+    if (logits_pshape.is_static())
+    {
+        if (logits_pshape[0].is_static())
+        {
+            batch_size = logits_pshape.to_shape()[0];
+            is_batch_size_set = true;
+        }
+        if (logits_pshape[1].is_static())
+        {
+            time_steps = logits_pshape.to_shape()[1];
+            is_time_steps_set = true;
+        }
+    }
+
+    if (logit_length_pshape.is_static())
+    {
+        if (is_batch_size_set)
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                logit_length_pshape[0].compatible(batch_size),
+                "The first dimension of logit length must be equal to the first dimension ",
+                "of the logits. Got: ",
+                logit_length_pshape[0],
+                " and: ",
+                batch_size);
+        }
+        else if (logit_length_pshape[0].is_static())
+        {
+            batch_size = logit_length_pshape.to_shape()[0];
+            is_batch_size_set = true;
+        }
+    }
+
+    if (labels_pshape.is_static())
+    {
+        if (is_batch_size_set)
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                labels_pshape[0].compatible(batch_size),
+                "The first dimension of labels must be equal to the first dimension ",
+                "of the logits and the logit length. Got: ",
+                labels_pshape[0],
+                " and: ",
+                batch_size);
+        }
+        else if (labels_pshape[0].is_static())
+        {
+            batch_size = labels_pshape.to_shape()[0];
+            is_batch_size_set = true;
+        }
+
+        if (is_time_steps_set)
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                labels_pshape[1].compatible(time_steps),
+                "The second dimension of labels must be equal to the second dimension ",
+                "of logits. Got: ",
+                labels_pshape[1],
+                " and: ",
+                time_steps);
+        }
+        else if (labels_pshape[1].is_static())
+        {
+            time_steps = labels_pshape.to_shape()[1];
+            is_time_steps_set = true;
+        }
+    }
+
+    if (label_length_pshape.is_static())
+    {
+        if (is_batch_size_set)
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                label_length_pshape[0].compatible(batch_size),
+                "The first dimension of label length must be equal to the first dimension ",
+                "of the logits, the logit length and labels. Got: ",
+                label_length_pshape[0],
+                " and: ",
+                batch_size);
+        }
+        else if (label_length_pshape[0].is_static())
+        {
+            batch_size = label_length_pshape.to_shape()[0];
```

Comment:
Maybe at the end of this code block we need add a check
```
NODE_VALIDATION_CHECK(
                this,
                label_length_pshape[0].compatible(batch_size),
                "The first dimension of label length must be equal to the first dimension ",
                "of the logits, the logit length and labels. Got: ",
                label_length_pshape[0],
                " and: ",
                batch_size);
```

Reply:
Fixed